src/Endatix.Core/UseCases/FormDefinitions/PartialUpdateActive/PartialUpdateActiveFormDefinitionHandler.cs

```diff
@@ -1,27 +1,49 @@
 ﻿using Endatix.Core.Abstractions.Repositories;
 using Endatix.Core.Entities;
+using Endatix.Core.Infrastructure.Domain;
 using Endatix.Core.Infrastructure.Messaging;
 using Endatix.Core.Infrastructure.Result;
 using Endatix.Core.Specifications;
 
 namespace Endatix.Core.UseCases.FormDefinitions.PartialUpdateActive;
 
-public class PartialUpdateActiveFormDefinitionHandler(IFormsRepository formRepository) : ICommandHandler<PartialUpdateActiveFormDefinitionCommand, Result<FormDefinition>>
+public class PartialUpdateActiveFormDefinitionHandler(IFormsRepository formsRepository, IRepository<Submission> submissionsRepository) : ICommandHandler<PartialUpdateActiveFormDefinitionCommand, Result<FormDefinition>>
 {
     public async Task<Result<FormDefinition>> Handle(PartialUpdateActiveFormDefinitionCommand request, CancellationToken cancellationToken)
     {
         var spec = new ActiveFormDefinitionByFormIdSpec(request.FormId);
-        var formWithActiveDefinition = await formRepository.SingleOrDefaultAsync(spec, cancellationToken);
+        var formWithActiveDefinition = await formsRepository.SingleOrDefaultAsync(spec, cancellationToken);
         var activeDefinition = formWithActiveDefinition?.ActiveDefinition;
         if (formWithActiveDefinition == null || activeDefinition == null)
         {
             return Result.NotFound("Active form definition not found.");
         }
 
-        activeDefinition.UpdateSchema(request.JsonData);
-        activeDefinition.UpdateDraftStatus(request.IsDraft);
+        if(await ShouldCreateNewFormDefinitionAsync(activeDefinition, request)) {
+            var newFormDefinition = new FormDefinition(request.IsDraft??false, request.JsonData);
+            formWithActiveDefinition.AddFormDefinition(newFormDefinition);
+            
+            if (!newFormDefinition.IsDraft) {
+                formWithActiveDefinition.SetActiveFormDefinition(newFormDefinition);
+            }
+           
+            await formsRepository.UpdateAsync(formWithActiveDefinition, cancellationToken);
```

Comment:
I see that this line will invoke alongside the other `UpdateAsync` on line 37. Is this intentional or probably can be removed?